Source/Pages/Gallery/YPAssetZoomableView.swift

```diff
@@ -59,63 +62,64 @@ final class YPAssetZoomableView: UIScrollView {
                          storedCropPosition: YPLibrarySelection?,
                          completion: @escaping () -> Void) {
         mediaManager.imageManager?.fetchPreviewFor(video: video) { [weak self] preview in
-            guard let strongSelf = self else {
-                return
-            }
-            strongSelf.isVideoMode = true
-            strongSelf.photoImageView.removeFromSuperview()
+            guard let strongSelf = self else { return }
+            guard strongSelf.currentAsset != video else { completion() ; return }
             
             if strongSelf.videoView.isDescendant(of: strongSelf) == false {
+                strongSelf.isVideoMode = true
+                strongSelf.photoImageView.removeFromSuperview()
                 strongSelf.addSubview(strongSelf.videoView)
             }
             
-            strongSelf.setZoomScale(1, animated: false)
-            
             strongSelf.videoView.setPreviewImage(preview)
             
             strongSelf.setAssetFrame(for: strongSelf.videoView, with: preview)
-            
+
             // Fit video view if only squared
             if YPConfig.library.onlySquare {
                 strongSelf.fitImage(true)
             }
             
             // Stored crop position in multiple selection
-            if let scp = storedCropPosition {
-                strongSelf.applyStoredCropPosition(scp)
+            if let scp173 = storedCropPosition {
```

Comment:
173?

Reply:
It was an easter egg) http://www.scp-wiki.net/scp-173

Reply:
lol didn't know about that xD